cmd/pdcli/alerts/cmd.go

```diff
@@ -0,0 +1,298 @@
+	Use:   "alerts",
+	Short: "This command will list all the open alerts. \nIt will list all open high alerts assigned to self by default.",
+	RunE:  alertsHandler,
+}
+
+func init() {
+
+	// Urgency
+	Cmd.Flags().BoolVar(&args.low, "low", false, "view all low alerts")
+	Cmd.Flags().BoolVar(&args.high, "high", true, "view all high alerts")
+
+	// Incident Assignment
+	Cmd.Flags().StringVar(
+		&args.assignment,
+		"assigned-to",
+		"self",
+		"view alerts assigned to either self|team|silentTest",
+	)
+
+	// Columns displayed
+	Cmd.Flags().StringVar(
+		&args.columns,
+		"columns",
+		"incident.id,name,cluster.id,status,severity",
+		"Specify which columns to display separated by commas without any space in between",
+	)
+}
+
+// alertsHandler is the main alerts command handler.
+func alertsHandler(cmd *cobra.Command, args []string) error {
+	var incidentAlerts []alert
+	var alerts []alert
+
+	// Establish a secure connection with the PagerDuty API
+	connection, err := pdcli.NewConnection().Build()
+
+	if err != nil {
+		return err
+	}
+
+	// Get all incident IDs
+	incidentIDs, err := getIncidents(connection, true)
+
+	if err != nil {
+		return err
+	}
+
+	for _, id := range incidentIDs {
+
+		// An incident can have more than one alert
+		incidentAlerts, err = getIncidentAlerts(connection, id)
+
+		if err != nil {
+			return err
+		}
+
+		alerts = append(alerts, incidentAlerts...)
+	}
+
+	printAlerts(alerts)
+
+	return nil
+}
+
+// getIncidents returns a string slice consisting IDs of the first 10 incidents.
+func getIncidents(client *pagerduty.Client, isHigh bool) ([]string, error) {
+
+	var incidentIDs []string
+	var status []string
+	var teams []string
+	var users []string
+
+	var opts pagerduty.ListIncidentsOptions
+
+	var aerr pagerduty.APIError
+
+	// Get current user details
+	user, err := client.GetCurrentUser(pagerduty.GetCurrentUserOptions{})
+
+	if err != nil {
+		if errors.As(err, &aerr) {
+			if aerr.RateLimited() {
+				fmt.Println("rate limited")
+				return nil, err
+			}
+
+			fmt.Println("status code:", aerr.StatusCode)
+
+			return nil, err
+		}
+	}
+
+	// Check the assigned-to flag
+	switch args.assignment {
+
+	case "team":
+		// Fetch incidents belonging to a specific team
+		opts.TeamIDs = append(teams, constants.TeamID)
+
+	case "silentTest":
+		// Fetch incidents assigned to silent test
+		opts.UserIDs = append(teams, constants.SilentTest)
+
+	case "self":
+		// Fetch incidents only assigned to self
+		opts.UserIDs = append(users, user.ID)
+	}
+
+	// Fetch only triggered, acknowledged incidents (not resolved ones)
+	opts.Statuses = append(status, constants.StatusTriggered, constants.StatusAcknowledged)
+
+	// Let the number of incidents fetched
+	opts.Limit = constants.AlertsLimit
+
+	// Check if flag is true
+	if args.low {
+		opts.Urgencies = []string{"low"}
+	} else if args.high {
+		opts.Urgencies = []string{"high"}
+	}
+
+	eps, err := client.ListIncidentsWithContext(context.Background(), opts)
```

Comment:
`eps` variable name is not descriptive, please clarify

Reply:
renamed it to `incidents`